python/lsst/daf/butler/core/ddl.py

```diff
@@ -281,6 +298,9 @@ def getSizedColumnType(self) -> sqlalchemy.types.TypeEngine:
             A SQLAlchemy column type object.
         """
         if self.length is not None:
+            # Last chance check that we are only looking at possible String
+            if self.dtype == sqlalchemy.String and not self.isStringType():
```

Comment:
I think `self.dtype is sqlalchemy.String` is safe here, and probably preferable.